4.analyze_data/notebooks/UMAP/scripts/0.UMAP_coordinates.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Generate UMAP coordinates for each plate
+
+# ## Import libraries
+
+# In[1]:
+
+
+import glob
+import pathlib
+import pandas as pd
+import umap
+
+from pycytominer import feature_select
+from pycytominer.cyto_utils import infer_cp_features
+
+
+# ## Set constants
+
+# In[2]:
+
+
+# Set constants
+umap_random_seed = 1234
+umap_n_components = 2
+
+output_dir = pathlib.Path("results")
+output_dir.mkdir(parents=True, exist_ok=True)
+
+
+# ## Create list of paths to feature selected data per plate
+
+# In[3]:
+
+
+# Set input paths
+data_dir = pathlib.Path("../../../3.processing_features/data/single_cell_profiles/")
+
+# Select only the feature selected files
+file_suffix = "*sc_feature_selected.parquet"
+
+# Obtain file paths for all feature selected plates
+fs_files = glob.glob(f"{data_dir}/{file_suffix}")
+fs_files
+
+
+# In[4]:
+
+
+# Load feature data into a dictionary, keyed on plate name
+cp_dfs = {x.split("/")[-1]: pd.read_parquet(x) for x in fs_files}
+
+# Print out useful information about each dataset
+print(cp_dfs.keys())
+[cp_dfs[x].shape for x in cp_dfs]
+
+
+# ## Generate UMAP coordinates for each plate
+# 
+# **Note:** Only metadata that is common between plates are included in final data frame.
+
+# In[5]:
+
+
+desired_columns = ["Metadata_Well", "Metadata_Site", "Metadata_number_of_singlecells", "Metadata_genotype"]
+
+# Fit UMAP features per dataset and save
+for plate in cp_dfs:
```

Comment:
nice use of a for loop and saving memory

Reply:
I got this from Erik so thanks to him that I learned this!